python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,612 @@
+            male_county_request_params = format_params(
+                HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES, True)
+            female_county_request_params = format_params(
+                HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX, HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES, True)
+
+            male_state_data = self.get_acs_data_from_variables(
+                male_state_request_params)
+            female_state_data = self.get_acs_data_from_variables(
+                female_state_request_params)
+            male_county_data = self.get_acs_data_from_variables(
+                male_county_request_params)
+            female_county_data = self.get_acs_data_from_variables(
+                female_county_request_params)
+
+        # Aggregate and accumulate data in memory
+        self.accumulate_acs_data(male_state_data)
+        self.accumulate_acs_data(female_state_data)
+        self.accumulate_acs_data(male_county_data)
+        self.accumulate_acs_data(female_county_data)
+
+    '''
+    Takes data in the form of
+    [
+        [C27001A_002E, C27001A_003E, C27001B_002E, C27001B_002E][state][?county],
+        [12345, 1245, 123546, 124567, 01, 02]
+        ...
+    ]
+    This method determines the variables at the top (aka: key_row)
+    matches the value with the metadata from the prefix_suffix list
+    and stores it in the self.data as a tuple
+
+    (state_fip, county_fip, age, sex, race) example:
+    {
+        "('01', None, '0-6', 'Male', None)": {
+            "Total": "177643",
+            "With": "172850",
+            "Without": "4793"
+        },
+        "('01', None, '6-18', 'Male', None)": {
+            "Total": "414407",
+            "With": "400460",
+            "Without": "13947"
+        },
+        ...
+    } Note: This can be debugged via the total_health_insurance.json output file via local_debug
+    '''
+
+    def accumulate_acs_data(self, data):
+        key_row = data[0]
+        for row in data[1::]:
+            data = {}
+            for key in key_row:
+                if(key != 'state' and key != 'county'):
+                    data[key] = {}
+            state_fip = None
+            county_fip = None
+            col_index = 0
+            for col in row:
+                key = key_row[col_index]
+                if(key == 'state'):
+                    state_fip = col  # Extract the static key_row state
+                elif key == 'county':
+                    # Extract the static key_row (county) *if exists
+                    county_fip = col
+                else:
+                    data[key] = {'value': col, 'meta': self.metadata[key]}
+
+                col_index += 1
+
+            for var in data:
+                metadata = data[var]['meta']
+                value = data[var]['value']
+                row = self.upsert_row(state_fip, county_fip, metadata.get(
```

Comment:
Rather than doing "metdata.get(...)" you should just be able to index directly, eg "metadata[...]"

Reply:
This is not the case, as race and sex are not gaurenteed to be set and will trigger a Key Error

Reply:
I see